crates/sp-domains/src/lib.rs

```diff
@@ -607,6 +609,12 @@ impl DomainsDigestItem for DigestItem {
     }
 }
 
+/// The storage key of the `SelfDomainId` storage item in the `pallet-domain-id`
+pub const SELF_DOMAIN_ID_STORAGE_KEY: [u8; 32] = [
```

Comment:
Please construct an actual storage and derive the key instead of magic bytes

Reply:
We can construct something `SelfDomainId::<T>::hashed_key().to_vec()` in the runtime but not the client side, I also consider runtime API but this key is used in the consensus node, thus we need to add the `pallet-domain-id` to consensus runtime to do this, I'm not sure if it is worth, so I just add a unit test to `pallet-domain-id` to ensure it won't changed accidentally.

Reply:
We already derive something similar here - https://github.com/subspace/subspace/blob/main/domains/primitives/messenger/src/messages.rs#L366 without a runtime call. Introducing this will alleviate the readability problem rather than seeing some magic bytes. Ofcourse, the storage definition on the runtime is directly connected, hence, this client side impl would need to be updated accordingly. I doubt we will change storage definitions that often and even if they client usage can follow the similar usage